sacred/stflow/internal.py

```diff
@@ -31,8 +31,8 @@ def __init__(self, classx, method_name, decorator_func):
     def __enter__(self):
 
         self.original_method = getattr(self.classx, self.method_name)
-        if not getattr(self.original_method,
-                       "sacred_patched%s" % self.__class__.__name__, False):
+        if not hasattr(self.original_method,
```

Comment:
This seems to make total sense. Maybe just keep this change and drop the one in `dependencies.py`